task_2.py

```diff
@@ -0,0 +1,91 @@
+class Product:
+    """Product class implementation"""
+
+    def __init__(self, name: str, price: float) -> None:
+        """Initialize a product instance"""
+
+        self.name = name
+        self.price = price
+
+    def __str__(self):
+        """Implement a human readable string representation for Product class"""
+
+        return f"{self.name}, {self.price})"
+
+    def __eq__(self, other) -> bool:
+        """Implement equality comparison for the Product instances. The products are equal in case both objects have the
+         same name and price."""
+
+        if not isinstance(other, Product):
+            return False
+        return self.name == other.name and self.price == other.price
+
+    def __float__(self) -> float:
+        """Implement type-casting for Product. Product cast to float type equals to its price"""
+
+        return float(self.price)
```

Comment:
```suggestion
        return self.price
```

Це не помилка, але з точку зору архітектури у тебе вже зазначено, що це атрибут float-типу. Отже не має необхідності додатко його якось перетворювати у float.